### DISCUSSION TITLE
Re-using actions from other repos (what is .github/actions for?)

### DISCUSSION THREAD

**Discussion by JonnyOThan** [community/community#116540] [Actions] [answered]
### Select Topic Area

Question

### Body

TLDR: I have a reusable action in `SomeUser/SomeRepo/.github/actions/build/action.yml` and I'm trying to call it from a workflow in a different repo with `uses: SomeUser/SomeRepo/build@main` but I get an error:

> Error: Can't find 'action.yml', 'action.yaml' or 'Dockerfile' for action 'SomeUser/SomeRepo/build@main'.

----

I have a repo with a bunch of common tools and files, and I thought I'd put some reusable actions in there.  This repo is NOT solely for hosting actions, even though I see that is "recommended."  If possible I'd like to keep it that way.

In [the documentation](https://docs.github.com/en/actions/creating-actions/about-custom-actions#choosing-a-location-for-your-action) it recommends placing such actions in the ".github/actions" directory: 

> If you plan to combine action, workflow, and application code in a single repository, we recommend storing actions in the .github directory. For example, .github/actions/action-a and .github/actions/action-b.

I can't find any other reference to this directory anywhere online, mostly because search tools (google, bing) seem to ignore the punctuation.

How do I reuse this action?  Should it *not* be in the .github/actions folder?  Do I need to include that in the `uses` path?

----

OK, so fully specifying the path to the action works: `uses: SomeUser/SomeRepo/.github/actions/build@main`

So what's the point of the `.github/actions` directory?

**Comment by ste-xx** [comment, marked as answer]
It's just a recommendation.
So if you managed many different builds, you can easily find the (local) build actions.

Standalone actions contains normally the action.yml in the root directory. 
Example: [https://github.com/actions/github-script](https://github.com/actions/github-script) 

So that you can call it like this:

```
- uses: actions/github-script@v7
```
